### PR TITLE
feat: implement `ServerNode` POST request method

### PR DESCRIPTION
**Pull request by sno2**
#7

### CODE REVIEW

File: src/server-node.ts [hunk 1]
```diff
@@ -30,8 +30,8 @@ export class ServerNode {
 
   /**
    * Used internally for handling paginated requests.
-   * @param endpoint The endpoint to send the request to.
-   * @param options The optional object for the pagination options. It can have a `limit` or `offset` key/value pair.
+   * @param endpoint the endpoint to send the request to
```
Comment (sno2) [inline src/server-node.ts:33]: I also fixed these formatting issues.

File: src/server-node.ts [hunk 2]
```diff
@@ -18,8 +18,8 @@ export class ServerNode {
 
   /**
    * Gets data for the given endpoint with the given query params.
-   * @param endpoint
-   * @param params
+   * @param endpoint the endpoint to send the request to
+   * @param params the optional object for the query params
```
Comment (sno2) [inline src/server-node.ts:22]: I also fixed these formatting issues.

**Review by sno2 — COMMENTED** [review]
I also fixed these formatting issues.

**Comment by lapstjup** [comment]
@sno2 Where does the declare code comes in ?

**Comment by sno2** [comment]
> @sno2 Where does the declare code come in ?

That is a type declaration and it defines the shape of the typescript code.

**Comment by lapstjup** [comment]
@sno2 I see. I was asking because I don't see it in this PR as you mentioned in the issue. All the declarations would be in one d.ts file I guess in the future ?

**Comment by sno2** [comment]
The d.ts files are all generated when you run `npm run build`.  Also, the client libraries only give the users back the actual body of the requests.  If there is an error while doing these requests, it will have to be done on the client side (currently).  This is definitely a design question we need to be asking ourselves, though.

**Comment by lapstjup** [comment]
@sno2 Thanks for the info. Btw the changes look good 😄.

**Pull request merged**